api/src/test/java/org/openmrs/EncounterTest.java

```diff
@@ -1262,4 +1264,43 @@ public void multipleAddingAndRemovingOfSameProvider_shouldNotFail() {
 
 	}
 
+	/**
+	 * @see Encounter#hasDiagnosis(Diagnosis)
+	 */
+	@Test
+	public void hasDiagnosis_shouldCheckIfEncounterHasDiagnosis(){
+		Encounter encounter = new Encounter();
+		Diagnosis diagnosis = new Diagnosis();
+		diagnosis.setEncounter(encounter);
+		diagnosis.setCondition(new Condition());
+		diagnosis.setCertainty(ConditionVerificationStatus.CONFIRMED);
+		diagnosis.setPatient(new Patient());
+		diagnosis.setRank(2);
+		
+		Set<Diagnosis> diagnoses = new HashSet<>();
+		diagnoses.add(diagnosis);
+		
+		encounter.setDiagnoses(diagnoses);
+		
+		Assert.assertTrue(encounter.hasDiagnosis(diagnosis));
+	}
+
+	/**
+	 * @see Encounter#hasDiagnosis(Diagnosis)
+	 */
+	@Test
+	public void hasDiagnosis_shouldCheckIfEncounterDoesNotHaveDiagnosis(){
```

Comment:
This should be renamed to hasDiagnosis_shouldReturnFalseIfEncounterDoesNotHaveDiagnosis

Reply:
Okay. Done